File: server/src/main/java/socialMediaApp/services/LikeService.java
package socialMediaApp.services;

import org.springframework.stereotype.Service;
import socialMediaApp.mappers.LikeMapper;
import socialMediaApp.models.Like;
import socialMediaApp.repositories.LikeRepository;
import socialMediaApp.requests.LikeRequest;
import socialMediaApp.responses.like.LikeResponse;

import java.util.List;
import java.util.Optional;

@Service
public class LikeService {

    private final LikeRepository likeRepository;
    private final LikeMapper likeMapper;

    public LikeService(LikeRepository likeRepository, LikeMapper likeMapper) {
        this.likeRepository = likeRepository;
        this.likeMapper = likeMapper;
    }

    public List<LikeResponse> getAllByPost(int postId){
        List<Like> likes = likeRepository.findAllByPost_Id(postId);
        return likeMapper.likesToLikeResponses(likes);
    }

    public List<LikeResponse> getAllByUser(int userId){
        List<Like> likes = likeRepository.findAllByUser_Id(userId);
        return likeMapper.likesToLikeResponses(likes);
    }

    public boolean isLiked(int userId,int postId){
        Optional<Like> like = likeRepository.findByUser_IdAndPost_Id(userId,postId);
        return like.isPresent();
    }

    public void add(LikeRequest likeRequest){
        if (isLiked(likeRequest.getUserId(), likeRequest.getPostId())){
            return;
        }
        Like like = likeMapper.requestToLike(likeRequest);
        likeRepository.save(like);
    }

    public void delete(LikeRequest likeRequest){
        Optional<Like> like = likeRepository.findByUser_IdAndPost_Id(likeRequest.getUserId(),likeRequest.getPostId());
       likeRepository.delete(like.get());
    }

}

File: client/src/services/LikeService.js
import axios from 'axios'

export default class LikeService{
    add(userId,postId,token){   
        const values = {userId,postId}
        return axios.post(process.env.REACT_APP_API+"likes/add",values,{
            headers:{
                'Authorization':"Bearer "+token
            }
        })
    }

    delete(userId,postId,token){
        const values = {userId,postId}
        return axios.post(process.env.REACT_APP_API+"likes/delete",values,{
            headers:{
                'Authorization':"Bearer "+token
            }
        })
    }

    isLiked(userId,postId,token){
        return axios.get(process.env.REACT_APP_API+`likes/isliked?userId=${userId}&postId=${postId}`,{
            headers:{
                'Authorization':"Bearer "+token
            }
        })
    }

    getLikesByPost(postId,token){
        return axios.get(process.env.REACT_APP_API+"likes/getallbypost/"+postId,{
            headers:{
                'Authorization':"Bearer "+token
            }
        })
    }


}